### DISCUSSION TITLE
Decorative-only text with -dmenu?

### DISCUSSION THREAD

**Discussion by cortsf** [davatorium/rofi#2009] [Q&A] [answered]
Hello. It's possible to declare somehow a "decorative" column to document each row in the listview? I.e: A non selectable column not producing any output and not getting selected when input matches text on this column. If not, it would be too difficult to add such a feature to rofi? Something like..

``` bash
echo "ls;Use this one to list files!!|ps;This one lists processes!!" | rofi -dmenu -sep '|' -decorative-column-separator ';'
```

It's for web browser search engines, ideally each row has to be a pair of `(<keyword/alias>, <url>)` where `<url>` is for the user to have some visual feedback. Input should match and return `<keyword/alias>` only.

**Comment by DaveDavenport** [comment]
you can make elements you cannot  'accept' (press return),  it is not possible to have them 'unselectable' as in you cannot highlight them.. this in current design is not possible.

would this not work well enough for the above example?

```bash
echo -en 'ls\n<i>Use this one to list files!!</i>|ps\n<i>This one lists processes!!</i>' | rofi -dmenu -sep '|' -eh 2 -markup-rows -no-config
```

![rofi-2024-07-17-2003-00000](https://github.com/user-attachments/assets/62adae25-528b-4a34-99ea-3d5a9d66c091)

**Comment by cortsf** [comment]
The auxiliary line `<i>..</i>` sent to stdout can be parsed out down the (pipe)line but I'm trying to avoid user input to match text on these strings, causing the rows to get highlighted. Here for example the first row (ls) will get selected/highlighted if the user writes "ps".

```
echo -en 'ls\n<i>Use this ps one to ps list files!!</i>|ps\n<i>This one lists processes!!</i>' | rofi -dmenu -sep '|' -eh 2 -markup-rows -no-config
```

Since this is for pairs of short keys (this is what the user needs) and long urls (doc only) i'm afraid that user input will always match and highlight irrelevant rows based on urls rather than keys 

``` lisp
("gm", "https://mail.google.com/mail/u/0/#search/{}")
("gh.rofi", "https://github.com/search?q=repo%3Adavatorium%2Frofi+{}&type=issues")
...
```

But I'll try to see how it wors. Thanks for the comment and for Rofi! Amazing tool. If there is any other possible solution would love to hear about it, I'm working on something that may be of public interest.

**Comment by cortsf** [comment, marked as answer]
Adding `-matching regex -filter "^"` to your snippet solves the problem.

```
echo -en 'ls\n<i>Use this psone to ps list files!!</i>|ps\n<i>This one lists processes!!</i>' | rofi -dmenu -sep '|' -eh 2 -markup-rows -no-config  -matching regex -filter '^'
```

---

My problem is solved already (Thanks @DaveDavenport ) but (just thinking) it would be a nice feature if rofi could wrap "plain" user imput with regexes declared using placeholders or something like that. For this trivial use case it could be something like:


- Regex template: `^__special_placeholder__`
- User input in rofi: `ps`
- String rofi uses under the hood: `^ps`

**Comment by DaveDavenport** [comment]
Did you check the manpage? https://davatorium.github.io/rofi/1.7.5/rofi-script.5/

in the next release some extra features are coming that might help you:
https://davatorium.github.io/rofi/current/rofi-script.5/

**Comment by cortsf** [comment]
I knew of script mode but my use case was fairly simple originally. But it worked so well, it's growing serious now and script mode might be the way to go. Look at this beauty! It's a rofi interface for qutebrowser. The code is pretty simple and there is no lag in calling `rofi -dmenu` recursively to interact with the user tbh, not sure what could script mode could offer other than looking more like the right way to do it.


![rofi](https://github.com/user-attachments/assets/471d0e94-d5b9-4f27-98c4-dc40785a92a4)

**Reply by DaveDavenport** [reply]
a lot of the script mode features work in dmenu..  See the rofi-dmenu manpage :D

F.e., in the next release you can display something different from what is searched/returned.
